nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/util/P2PathUtils.java

```diff
@@ -159,4 +167,18 @@ public P2Attributes toP2AttributesBinary(final TokenMatcher.State state) {
         .path(binaryPath(path(state), name(state), version(state)))
         .build();
   }
+
+  public static String escapeUriToPath(final String uri) {
+    return uri.replace("://", "/");
+  }
+
+  public static String unescapePathToUri(final String path) {
+    if (path.startsWith(HTTP_NXRM_PREFIX)) {
+      return path.replaceFirst(HTTP_NXRM_PREFIX, HTTP_URL_PREFIX);
```

Comment:
Question: Is it possible to make one outgoing point (one 'return')?

Reply:
Done